### PR TITLE
[SPARK-19292][SQL] filter with partition columns should be case-insensitive on Hive tables

### PR DESCRIPTION
**Pull request by cloud-fan**
## What changes were proposed in this pull request?

When we query a table with a filter on partitioned columns, we will push the partition filter to the metastore to get matched partitions directly.

In `HiveExternalCatalog.listPartitionsByFilter`, we assume the column names in partition filter are already normalized and we don't need to consider case sensitivity. However, `HiveTableScanExec` doesn't follow this assumption. This PR fixes it.

## How was this patch tested?

new regression test

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @yhuai @gatorsmile @ericl

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala [hunk 1]
```diff
@@ -2014,4 +2014,17 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
...
     }
   }
+
+  test("SPARK-19292: filter with partition columns should be case-insensitive on Hive tables") {
+    withTable("tbl") {
+      sql("CREATE TABLE tbl(i int, j int) USING hive PARTITIONED BY (j)")
+      sql("INSERT INTO tbl PARTITION(j=10) SELECT 1")
+      checkAnswer(spark.table("tbl"), Row(1, 10))
+
+      sql("SELECT i, j FROM tbl WHERE J=10").explain(true)
```
Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: Nit: remove it?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala [hunk 2]
```diff
@@ -2014,4 +2014,17 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
       )
     }
   }
+
+  test("SPARK-19292: filter with partition columns should be case-insensitive on Hive tables") {
+    withTable("tbl") {
```
Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: Could you explicitly set the conf `CASE_SENSITIVE` to false? For example
```scala
    withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false") {
      ... 
    }

```

**Comment by gatorsmile** [comment]
LGTM, except two comments.

**Comment by gatorsmile** [comment]
LGTM waiting for tests

**Comment by gatorsmile** [comment]
Thanks! Merging to master

**Pull request closed without merging**